src/cpp/fastdds/xtypes/type_conversion/dyn_type_tree.ipp

```diff
@@ -0,0 +1,1116 @@
+        array_str += "[";
+        array_str += std::to_string(bound);
+        array_str += "]";
+    }
+
+    return ret;
+}
+
+ReturnCode_t map_kind_to_str(
+        const DynamicType::_ref_type& dyn_type,
+        std::string& map_str) noexcept
+{
+    if (dyn_type->get_kind() != TK_MAP)
+    {
+        EPROSIMA_LOG_ERROR(DYN_TYPES, "Type is not a map.");
+        return RETCODE_BAD_PARAMETER;
+    }
+
+    ReturnCode_t ret = RETCODE_OK;
+
+    TypeDescriptor::_ref_type type_descriptor {traits<TypeDescriptor>::make_shared()};
+    ret = dyn_type->get_descriptor(type_descriptor);
+
+    if (ret != RETCODE_OK)
+    {
+        return ret;
+    }
+
+    std::string key_str;
+    const auto key_type = type_descriptor->key_element_type();
+    ret = type_kind_to_str(key_type, key_str);
+
+    if (ret != RETCODE_OK)
+    {
+        return ret;
+    }
+
+    std::string value_str;
+    const auto value_type = type_descriptor->element_type();
+    ret = type_kind_to_str(value_type, value_str);
+
+    if (ret != RETCODE_OK)
+    {
+        return ret;
+    }
+
+    std::stringstream ss;
+
+    ss << "map<" << key_str << ", " << value_str << ">";
+
+    map_str = ss.str();
+
+    return ret;
+}
+
+ReturnCode_t sequence_kind_to_str(
+        const DynamicType::_ref_type& dyn_type,
+        std::string& sequence_str) noexcept
+{
+    if (dyn_type->get_kind() != TK_SEQUENCE)
+    {
+        EPROSIMA_LOG_ERROR(DYN_TYPES, "Type is not a sequence.");
+        return RETCODE_BAD_PARAMETER;
+    }
+
+    ReturnCode_t ret = RETCODE_OK;
+
+    DynamicType::_ref_type internal_type;
+    ret = container_internal_type(dyn_type, internal_type);
+
+    if (ret != RETCODE_OK)
+    {
+        return ret;
+    }
+
+    ret = type_kind_to_str(internal_type, sequence_str);
+
+    if (ret != RETCODE_OK)
+    {
+        return ret;
+    }
+
+    sequence_str = "sequence<" + sequence_str;
+
+    BoundSeq bounds;
+    ret = container_size(dyn_type, bounds);
+
+    if (ret != RETCODE_OK)
+    {
+        return ret;
+    }
+
+    for (const auto& bound : bounds)
+    {
+        const auto UNBOUNDED = static_cast<std::uint32_t>(LENGTH_UNLIMITED);
+
+        if (bound != UNBOUNDED)
+        {
+            sequence_str += ", " + std::to_string(bound);
+        }
+    }
+
+    sequence_str += ">";
+
+    return ret;
+}
+
+ReturnCode_t get_members_sorted(
+        const DynamicType::_ref_type& dyn_type,
+        std::vector<std::pair<std::string, MemberDescriptor::_ref_type>>& result) noexcept
+{
+    ReturnCode_t ret = RETCODE_OK;
+
+    // Skip the inherited members
+    TypeDescriptor::_ref_type type_descriptor {traits<TypeDescriptor>::make_shared()};
+    ret = dyn_type->get_descriptor(type_descriptor);
+
+    if (ret != RETCODE_OK)
+    {
+        return ret;
+    }
+
+    const auto base_type = type_descriptor->base_type();
+
+    std::uint32_t first_member = 0;
+
+    if (base_type != nullptr)
+    {
+        // If the struct has a base type, the first member is the base type
```

Comment:
Please rephrase, I understand what you are doing but I find the sentence misleading.

Reply:
And add a note/warning saying that it is assumed this is the case (base members are always before derived ones).

Reply:
I have rephrased the sentence, but I don't think a note or warning are necessary since it's part of the standard. In Chat-GPT's words: "According to the xtypes specification, when you derive one type from another, the base class members must indeed have lower indices than the members of the derived class. "